### DISCUSSION TITLE
Is Terminal configuration window necessary  ?

### DISCUSSION THREAD

**Discussion by TheNicker** [doublecmd/doublecmd#1062] [General]
I've just created a new shortcut `Shift+F9` - Run terminal with elevated privileges (Run as Administrator) so I won't have to be running double commander itself with elevated permissions. so far it works great. 

Then it got me wondering whether we really need the terminal configuration window
It seems like it's just a specialization of shell execution and there are only 3 options `normal/close after/stay open` and as you can see non of those runs my new command.

There are many shells both for windows and Linux and they can run with different parameters, so permutations are numerous.

Is it really necessary or **can we get rid of the terminal configuration window** and generalize terminal execution  ? 

![image](https://github.com/doublecmd/doublecmd/assets/4753283/4f97d83e-e950-4c51-945b-6b7af8c3e1a3)

**Comment by alexx2000** [comment]
Command can be one while parameters not. DC use they in the different situations. Really DC needs five different terminal settings.

**Comment by Skif-off** [comment]
@alexx2000 

> Really DC needs five different terminal settings.

What do you mean?

**Comment by TheNicker** [comment]
@alexx2000 
> Command can be one while parameters not

Yes indeed, however I still don't understand if we need specialization for Terminal commands in general.

**Comment by Skif-off** [comment]
@TheNicker, you can specify any command in each case and these commands can be different, they do not have to match.

@alexx2000, do you mean `{!TERM*}`?

**Reply by TheNicker** [reply]
@Skif-off, I mean we can have general purpose commands, that you can run anything not just terminal.
Then you can define predefined commands to run 100 terminals not just 5

**Comment by alexx2000** [comment]
@Skif-off, I mean:

1. Execute terminal
2. Execute command and close
3. Execute command and stay open
4. Execute file and close
5. Execute file and stay open

> Command can be one while parameters not.

Really not, macOS use different commands.

**Comment by alexx2000** [comment]
@TheNicker, they all are used at the specific places:

1. Execute terminal - used by F9 or from menu
2. Execute command and close - used by `{!TERMANDCLOSE}` macro
3. Execute command and stay open - used by `{!TERMSTAYOPEN}` macro and by Shift+Enter in the command line

**Comment by Skif-off** [comment]
@alexx2000 

> ... "Execute command" ... "Execute file" ...

I.e. DC will automatically select action for "file with parameters" or "file"?

> 4. Execute file and close

I.e. we can specify, for example, `xdg-open` instead of any terminal emulator?
(`ShellExecute` from WinAPI have the "open" action, but in non-Windows OS, the file must have execute attribute for current user.)

**Comment by alexx2000** [comment]
> I.e. we can specify, for example, xdg-open instead of any terminal emulator?

No.

> I.e. DC will automatically select action for "file with parameters" or "file"?

There are two cases:

1. Execute command in the terminal by Shift+Enter from command line 
2. Execute file in the terminal by Shift+Enter or via internal file associations

In the first case command should be passed as is while in the second case file name should be escaped.
